runtime/v2/shim/util_unix.go

```diff
@@ -63,20 +66,20 @@ func AdjustOOMScore(pid int) error {
 	return nil
 }
 
-// SocketAddress returns an abstract socket address
+// SocketAddress returns a socket address
 func SocketAddress(ctx context.Context, id string) (string, error) {
 	ns, err := namespaces.NamespaceRequired(ctx)
 	if err != nil {
 		return "", err
 	}
 	d := sha256.Sum256([]byte(filepath.Join(ns, id)))
-	return filepath.Join(string(filepath.Separator), "containerd-shim", fmt.Sprintf("%x.sock", d)), nil
+	return filepath.Join("/run/containerd/s", fmt.Sprintf("%x", d)), nil
```

Comment:
This breaks running multiple daemons

Reply:
Let's put runtime dir string into the hash

Reply:
ok, let me see if I can get that data easily.

Reply:
Also could you:
- add a code comment to explain that `/run/containerd/s` is hard coded for the sake of avoiding hitting the SUN_PATH length limit
- add `MkdirAll("/run/containerd/s")`

Reply:
> Let's put runtime dir string into the hash

We can also consider putting the daemon UUID into the hash, not sure which one is better.